enricher/standard/src/main/java/io/fabric8/maven/enricher/standard/HealthCheckEnricher.java

```diff
@@ -0,0 +1,109 @@
+package io.fabric8.maven.enricher.standard;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.extensions.DaemonSet;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.kubernetes.api.model.extensions.PetSet;
+import io.fabric8.maven.core.config.ProbeConfig;
+import io.fabric8.maven.core.handler.ProbeHandler;
+import io.fabric8.maven.enricher.api.AbstractHealthCheckEnricher;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by matthew.costa on 26/10/2016.
+ */
+public class HealthCheckEnricher extends BaseEnricher {
```

Comment:
I'm  not sure whether we should do this as an enricher, or as core part of the `ResourceMojo` plugin. The reason is, that the order of enricher matters, who will 'win', (see e.g. the more specialized enricher like `SpringBootHealthCheckEnricher`). If this functionality were independent of an enricher, then we can e.g. say: "A health check given in the configuration takes always precedence". Of course we can configure enrichers that way that this enricher has the highest priority, but that can easily break.

Another issue is, if you configure a health-check in the plugin configuration but *disable* this enricher (e.g. because of using another profile or exclude it in the enricher config) then this can lead to surprises, because someone would expect that its liveness  / readiness check configured in XML would be respected for sure.

In general I think that the code evaluating XML configuration from the plugin should go into the ResourceMojo (which tbh needs a refactoring). Said that, the XML configuration is not yet really very matured, but this is one of the next steps.